packages/builder/builder-webpack-provider/src/config/validate/experiments.ts

```diff
@@ -2,4 +2,8 @@ import { sharedExperimentsConfigSchema, z } from '@modern-js/builder-shared';
 import type { ExperimentsConfig } from '../../types';
 
 export const experimentsConfigSchema: z.ZodType<ExperimentsConfig> =
-  sharedExperimentsConfigSchema;
+  sharedExperimentsConfigSchema
+    .extend({
+      sourceBuild: z.boolean(),
```

Comment:
If the `sourceBuild` config is built in, we should also make `builder-plugin-source-build` as a built-in plugin.

But `builder-plugin-source-build` now has some third-party dependencies, we do not want to increase the install size of builder, so we need to reduce the dependencies if we want to build in it.

Reply:
I think some third-party dependencies should be required, such as @pnpm/filter-workspace-packages, @pnpm/logger. So we turn `builder-plugin-source-build` into an external plugin?

Reply:
I have removed some third-party dependencies from builder-plugin-source-build

Reply:
Yes, we can turn `builder-plugin-source-build` into an external plugin now.